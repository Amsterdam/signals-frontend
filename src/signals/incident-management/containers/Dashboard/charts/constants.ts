// SPDX-License-Identifier: MPL-2.0
// Copyright (C) 2023 Gemeente Amsterdam

export const timeFormatLocale = {
  dateTime: '%a %e %B %Y %X',
  date: '%d-%m-%Y',
  time: '%H:%M:%S',
  periods: ['AM', 'PM'],
  days: [
    'zondag',
    'maandag',
    'dinsdag',
    'woensdag',
    'donderdag',
    'vrijdag',
    'zaterdag',
  ],
  shortDays: ['Zo', 'Ma', 'Di', 'Wo', 'Do', 'Vr', 'Za'],
  months: [
    'januari',
    'februari',
    'maart',
    'april',
    'mei',
    'juni',
    'juli',
    'augustus',
    'september',
    'oktober',
    'november',
    'december',
  ],
  shortMonths: [
    'jan',
    'feb',
    'mrt',
    'apr',
    'mei',
    'jun',
    'jul',
    'aug',
    'sep',
    'okt',
    'nov',
    'dec',
  ],
}
