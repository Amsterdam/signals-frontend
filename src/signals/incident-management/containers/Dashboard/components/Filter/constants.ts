// SPDX-License-Identifier: MPL-2.0
// Copyright (C) 2023 Gemeente Amsterdam
export const filterNames = [
  'department',
  'category',
  'priority',
  'punctuality',
  'district',
]
