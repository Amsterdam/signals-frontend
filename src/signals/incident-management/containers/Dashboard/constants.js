export const REQUEST_DASHBOARD = 'src/Dashboard/REQUEST_DASHBOARD';
export const REQUEST_DASHBOARD_SUCCESS = 'src/Dashboard/REQUEST_DASHBOARD_SUCCESS';
