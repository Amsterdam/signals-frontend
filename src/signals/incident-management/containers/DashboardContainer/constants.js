export const REQUEST_DASHBOARD = 'src/DashboardContainer/REQUEST_DASHBOARD';
export const REQUEST_DASHBOARD_SUCCESS = 'src/DashboardContainer/REQUEST_DASHBOARD_SUCCESS';
export const REQUEST_DASHBOARD_ERROR = 'src/DashboardContainer/REQUEST_DASHBOARD_ERROR';
