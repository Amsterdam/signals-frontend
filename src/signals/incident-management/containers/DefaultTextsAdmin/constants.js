export const FETCH_DEFAULT_TEXTS = 'src/DefaultTextsAdmin/FETCH_DEFAULT_TEXTS';
export const FETCH_DEFAULT_TEXTS_SUCCESS = 'src/DefaultTextsAdmin/FETCH_DEFAULT_TEXTS_SUCCESS';
export const FETCH_DEFAULT_TEXTS_ERROR = 'src/DefaultTextsAdmin/FETCH_DEFAULT_TEXTS_ERROR';

export const STORE_DEFAULT_TEXTS = 'src/DefaultTextsAdmin/STORE_DEFAULT_TEXTS';
export const STORE_DEFAULT_TEXTS_SUCCESS = 'src/DefaultTextsAdmin/STORE_DEFAULT_TEXTS_SUCCESS';
export const STORE_DEFAULT_TEXTS_ERROR = 'src/DefaultTextsAdmin/STORE_DEFAULT_TEXTS_ERROR';

export const ORDER_DEFAULT_TEXTS = 'src/DefaultTextsAdmin/ORDER_DEFAULT_TEXTS';
