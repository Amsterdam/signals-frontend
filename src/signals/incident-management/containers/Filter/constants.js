export const SAVE_FILTER = 'sia/Filter/SAVE_FILTER';
export const SAVE_FILTER_SUCCESS = 'sia/Filter/SAVE_FILTER_SUCCESS';
export const SAVE_FILTER_FAILED = 'sia/Filter/SAVE_FILTER_FAILED';

export const UPDATE_FILTER = 'sia/Filter/UPDATE_FILTER';
export const UPDATE_FILTER_SUCCESS = 'sia/Filter/UPDATE_FILTER_SUCCESS';
export const UPDATE_FILTER_FAILED = 'sia/Filter/UPDATE_FILTER_FAILED';

export const GET_FILTERS = 'sia/Filter/GET_FILTERS';
export const GET_FILTERS_SUCCESS = 'sia/Filter/GET_FILTERS_SUCCESS';
export const GET_FILTERS_FAILED = 'sia/Filter/GET_FILTERS_FAILED';

export const CLEAR_FILTER = 'sia/Filter/CLEAR_FILTER';
