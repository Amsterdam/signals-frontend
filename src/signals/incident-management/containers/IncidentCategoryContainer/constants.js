export const REQUEST_CATEGORY_UPDATE = 'sia/IncidentCategoryContainer/REQUEST_CATEGORY_UPDATE';
export const REQUEST_CATEGORY_UPDATE_SUCCESS = 'sia/IncidentCategoryContainer/REQUEST_CATEGORY_UPDATE_SUCCESS';
export const REQUEST_CATEGORY_UPDATE_ERROR = 'sia/IncidentCategoryContainer/REQUEST_CATEGORY_UPDATE_ERROR';
