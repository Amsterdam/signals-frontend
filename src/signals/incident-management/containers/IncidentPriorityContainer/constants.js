export const REQUEST_PRIORITY_UPDATE = 'src/IncidentStatusContainer/REQUEST_PRIORITY_UPDATE';
export const REQUEST_PRIORITY_UPDATE_SUCCESS = 'src/IncidentStatusContainer/REQUEST_PRIORITY_UPDATE_SUCCESS';
export const REQUEST_PRIORITY_UPDATE_ERROR = 'src/IncidentStatusContainer/REQUEST_PRIORITY_UPDATE_ERROR';
