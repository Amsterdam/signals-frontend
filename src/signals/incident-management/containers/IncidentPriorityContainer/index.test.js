import React from 'react';
import { shallow } from 'enzyme';

import { IncidentPriorityContainer, mapDispatchToProps } from './index';
import { REQUEST_PRIORITY_UPDATE } from './constants';

jest.mock('./components/Add', () => 'Add');

describe('<IncidentPriorityContainer />', () => {
  let props;

  beforeEach(() => {
    props = {
      id: '1',
      incidentPriorityContainer: { subpriorityList: [] },
      onRequestPriorityUpdate: jest.fn(),
    };
  });

  afterEach(() => {
    jest.resetAllMocks();
  });

  it('should render correctly', () => {
    const wrapper = shallow(
      <IncidentPriorityContainer {...props} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  describe('mapDispatchToProps', () => {
    const dispatch = jest.fn();

    it('should request the priority update', () => {
      mapDispatchToProps(dispatch).onRequestPriorityUpdate({});
      expect(dispatch.mock.calls[0][0]).toEqual({ type: REQUEST_PRIORITY_UPDATE, payload: {} });
    });
  });
});
