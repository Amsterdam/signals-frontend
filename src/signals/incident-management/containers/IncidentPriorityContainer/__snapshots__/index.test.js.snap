// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IncidentPriorityContainer /> should render correctly 1`] = `
<div
  className="col-6"
>
  <div
    className="incident-edit-container"
  >
    <Component
      id="1"
      onRequestPriorityUpdate={[MockFunction]}
    />
  </div>
</div>
`;
